src/time_int.rs

```diff
@@ -31,27 +35,33 @@ pub trait TimeInt:
     /// // the result is not rounded, but truncated
     /// assert_eq!(8_u32.checked_mul_period(&Period::new(1,3)), Some(2_u32));
     /// ```
-    fn checked_mul_period(&self, period: &Period) -> Option<Self> {
-        Some(<Self as num::CheckedDiv>::checked_div(
-            &<Self as num::CheckedMul>::checked_mul(&self, &(*period.numerator()).into())?,
+    fn checked_mul_period<E: Error>(&self, period: &Period) -> Result<Self, TimeError<E>> {
+        <Self as num::CheckedDiv>::checked_div(
+            &<Self as num::CheckedMul>::checked_mul(&self, &(*period.numerator()).into())
+                .ok_or(TimeError::WouldOverflow)?,
             &(*period.denominator()).into(),
-        )?)
+        )
+        .ok_or(TimeError::WouldDivByZero)
     }
```

Comment:
It makes it necessary to add a generic to any function that might be used by a function that returns a Result<_, TimeError<E>> where E is the ImplError associated type in `Clock`.